meson.build

```diff
@@ -0,0 +1,177 @@
+project('waltham', 'c',
+        version: '0.0.0',
+        license: ['MIT'],
+        meson_version: '>=0.41.0')
+version = meson.project_version()
+desc = 'Network IPC library designed to resemble Wayland'
+
+pkgconfig = import('pkgconfig')
+python2 = find_program('python2')
+gen_py = find_program('tools/gen.py')
+
+libwaltham_src = [
+    'src/waltham/demarshaller.h',
+    'src/waltham/marshaller.c',
+    'src/waltham/marshaller.h',
+    'src/waltham/marshaller_log.h',
+    'src/waltham/message.c',
+    'src/waltham/message.h',
+    'src/waltham/waltham-connection.c',
+    'src/waltham/waltham-connection.h',
+    'src/waltham/waltham-object.c',
+    'src/waltham/waltham-object.h',
+    'src/waltham/waltham-private.h',
+    'src/waltham/waltham-util.c',
+    'src/waltham/waltham-util.h',
+]
+
+libwaltham_headers = [
+    'src/waltham/waltham-connection.h',
+    'src/waltham/waltham-object.h',
+    'src/waltham/waltham-util.h',
+]
+
+protocols_src = [
+    'data/private.xml',
+    'data/public.xml',
+    'data/command.xml',
+]
+
+client_h = custom_target('waltham-client.h',
+                         input: ['src/waltham/header-preamble.txt', protocols_src],
+                         output: 'waltham-client.h',
+                         depend_files: 'tools/gen.py',
+                         command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'client', '-t', 'header'],
+                         install: true,
+                         install_dir: 'include/waltham')
```

Comment:
Instead of specifying the install here, I think `client_h` would be an argument to `install_headers()`. Did you try that?
Maybe simply add it into the libwaltham_headers array if you move it below the custom targets.

Reply:
I did try that, but `install_headers()` didn’t like having a `custom_target` object instead of a string, so I reverted to that.  Maybe it’s a missing Meson feature, I’m not sure.

Reply:
Ok, let's keep it your way for now. It's quite possible it has been fixed in a later Meson, but OTOH our ultimate goal is to not install generated headers at all.